src/GAP/iso_gap_oscar.jl

```diff
@@ -180,6 +183,46 @@ function _iso_gap_oscar_number_field(FG::GapObj)
         coeffs = Vector{QQFieldElem}(GAPWrap.ExtRepOfObj(x))
         return FO(coeffs)
      end
+   elseif GAPWrap.IsNumberFieldByMatrices(FG)
+     # `pol` is the minimal polynomial of `M` because the GAP code
+     # caches the objects.
+     M = GAPWrap.PrimitiveElement(FG)
+     pol = GAPWrap.DefiningPolynomial(FG)
+
+     # Construct the isomorphic number field in Oscar.
+     cfs = GAPWrap.CoefficientsOfUnivariatePolynomial(pol)
+     R, x = polynomial_ring(QQ, "x")
```

Comment:
```suggestion
     R, x = polynomial_ring(QQ, "x", cached = false)
```

Reply:
Why shall we force the creation of a new polynomial ring whenever a new number field in Oscar gets created?

Reply:
Do we have a written down policy when to enable the caching and when not? @fieker?